nixos/modules/config/sysctl.nix

```diff
@@ -34,14 +33,373 @@ in
         options = {
           "net.core.rmem_max" = mkOption {
             type = types.nullOr highestValueType;
-            default = null;
             description = "The maximum receive socket buffer size in bytes. In case of conflicting values, the highest will be used.";
+            default = null;
           };
 
           "net.core.wmem_max" = mkOption {
             type = types.nullOr highestValueType;
-            default = null;
             description = "The maximum send socket buffer size in bytes. In case of conflicting values, the highest will be used.";
+            default = null;
+          };
+
+          "dev.tty.ldisc_autoload" = mkOption {
+            type = types.nullOr types.bool;
+            description = "Automatically load line discipline modules for tty devices.";
+            # Prevent unprivileged attackers from loading vulnerable line disciplines with the TIOCSETD ioctl
+            default = false;
+          };
+
+          "fs.protected_hardlinks" = mkOption {
+            type = types.nullOr types.bool;
+            description = "Protect hardlinks created by other users.";
+            # Prevent unprivileged attackers from creating world-writable sticky hardlinks
+            default = true;
+          };
+
+          "fs.protected_symlinks" = mkOption {
+            type = types.nullOr types.bool;
+            description = "Protect symlinks created by other users.";
+            # Prevent unprivileged attackers from creating world-writable sticky symlinks
+            default = true;
+          };
+
+          "fs.protected_fifos" = mkOption {
+            type = types.nullOr highestValueType;
+            description = "Protect pipes and FIFOs from being written to by other users.";
+            # Prevent unprivileged attackers from creating world-writable sticky FIFOs
+            default = 2;
+          };
+
+          "fs.protected_regular" = mkOption {
+            type = types.nullOr highestValueType;
+            description = "Protect files created by other users.";
+            # Prevent unprivileged attackers from creating world-writable sticky files
+            default = 2;
+          };
+
+          "fs.suid_dumpable" = mkOption {
+            type = types.nullOr highestValueType;
+            description = "Controls the core dump mode for setuid processes.";
+            # Prevent setuid processes from dumping core
+            default = 0;
+          };
+
+          "kernel.ctrl-alt-del" = mkOption {
+            type = types.nullOr types.bool;
+            description = "Enable the Ctrl-Alt-Del key combination to trigger a kernel reboot.";
+            # This is a security risk, as it allows users to reboot the system without authentication
+            default = false;
```

Comment:
But `services.logind.powerKey` still defaults to `poweroff`, leaving a possibility for the unprivileged user to shut down the system. Is there a need for this one?

Reply:
No, since the default is false anyway. I'd prefer to leave this in anyway, just so that it appears at https://search.nixos.org/options

Reply:
Hmm, `sysctl` gave me 0 too, but C-A-Delete still reboot the PC. I wonder if logind is the reason

Reply:
On NixOS w/ `systemd`, C-A-Del triggers `/etc/systemd/system/ctrl-alt-del.target`.